src/store/level-wrapper.ts

```diff
@@ -170,6 +172,42 @@ export class LevelWrapper<V> {
     return abortOr(options?.signal, this.db.batch(operations));
   }
 
+  private async compactUnderlyingStorage(options?: LevelWrapperOptions): Promise<void> {
+    options?.signal?.throwIfAborted();
+
+    await abortOr(options?.signal, this.createLevelDatabase());
+
+    const range = this.sublevelRange;
+    if (!range) {
+      return;
+    }
+
+    // additional methods are only available on the root API instance
+    const root = this.root;
+
+    if (root.db.supports.additionalMethods.compactRange) {
+      return abortOr(options?.signal, root.db['compactRange']?.(...range));
+    }
+  }
+
+  private get sublevelRange(): [string, string] | undefined {
+    const prefix = this.db['prefix'] ?? '';
```

Comment:
For my learning/understanding: is `prefix` an implementation specific detail of sublevel?

Reply:
yeah `prefix` is only a thing for `Sublevel`